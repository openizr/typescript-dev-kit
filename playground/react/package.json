{
  "license": "MIT",
  "devDependencies": {
    "@types/prop-types": "^15.7.4",
    "@types/react": "^17.0.26",
    "@types/react-dom": "^17.0.9",
    "typescript-dev-kit": "file:../../../../typescript-dev-kit/dist"
  },
  "dependencies": {
    "basx": "^1.3.7",
    "diox": "^4.0.6",
    "prop-types": "^15.7.2",
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
  },
  "eslintConfig": {
    "extends": [
      "./node_modules/typescript-dev-kit/main.js"
    ]
  },
  "tsDevKitConfig": {
    "target": "web",
    "devServer": {
      "ip": "0.0.0.0",
      "port": 5001
    },
    "html": {
      "template": "./html/index.html",
      "entries": [
        "sass",
        "typescript"
      ]
    },
    "entry": {
      "sass": "./scripts/sass.ts",
      "typescript": "./scripts/typescript.tsx",
      "javascript": "./scripts/javascript.jsx"
    },
    "splitChunks": true,
    "srcPath": "src",
    "distPath": "public",
    "banner": "/*!\n * Copyright (c) Matthieu Jabbour.\n * All rights reserved.\n */",
    "env": {
      "development": {
        "ENV": "development",
        "NODE_ENV": "development"
      },
      "production": {
        "ENV": "production",
        "NODE_ENV": "production"
      }
    }
  },
  "scripts": {
    "init": "node_modules/typescript-dev-kit/scripts/init.sh",
    "dev": "cd node_modules/typescript-dev-kit/ && node scripts/dev",
    "build": "cd node_modules/typescript-dev-kit/ && node scripts/build && cd ../../ && sed -i -e \"s%/\\*WEBPACK_CSS\\*/%WEBPACK_CSS\\n%g\" public/index.html  && sed -i -e \"/WEBPACK_CSS/r $(ls public/assets/styles/common.*.css)\" public/index.html && sed -i -e \"s/WEBPACK_CSS//g\" public/index.html",
    "test": "cd node_modules/typescript-dev-kit/ && node scripts/test",
    "doc": "node_modules/.bin/typedoc --out ./docs/ --exclude \"**/__+(tests|mocks)__/**\" src/scripts/javascript.jsx",
    "postinstall": "rm -f node_modules/.eslintcache"
  }
}
