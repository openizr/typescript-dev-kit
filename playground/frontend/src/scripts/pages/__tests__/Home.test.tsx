/**
 * @jest-environment jsdom
 */

import React from 'react';
import Home from 'scripts/pages/Home';
import { render } from '@testing-library/react';

describe('react/Home', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('renders correctly - basic', () => {
    const { container } = render(<Home translate={(label: string): string => label} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
