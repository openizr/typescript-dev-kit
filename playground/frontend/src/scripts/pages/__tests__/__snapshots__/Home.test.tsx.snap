// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/Home renders correctly - basic 1`] = `
<div>
  <p>
    <img
      alt="test"
      src="file"
    />
    <img
      alt="test"
      src="file"
    />
    LABEL_TEST
  </p>
</div>
`;
